src/System.Memory/src/System/Buffers/ReadOnlyBuffer.cs

```diff
@@ -0,0 +1,417 @@
+            BufferStart = new SequencePosition(segment, 0 | MemoryListStartMask);
+            BufferEnd = new SequencePosition(segment, data.Length | MemoryListEndMask);
+        }
+
+        public ReadOnlyBuffer(IEnumerable<Memory<T>> buffers)
+        {
+            ReadOnlyBufferSegment segment = null;
+            ReadOnlyBufferSegment first = null;
+            foreach (var buffer in buffers)
+            {
+                var newSegment = new ReadOnlyBufferSegment()
+                {
+                    Memory = buffer,
+                    RunningIndex = segment?.RunningIndex ?? 0
+                };
+
+                if (segment != null)
+                {
+                    segment.Next = newSegment;
+                }
+                else
+                {
+                    first = newSegment;
+                }
+
+                segment = newSegment;
+            }
+
+            if (first == null)
+            {
+                first = segment = new ReadOnlyBufferSegment();
+            }
+
+            BufferStart = new SequencePosition(first, 0 | MemoryListStartMask);
+            BufferEnd = new SequencePosition(segment, segment.Memory.Length | MemoryListEndMask);
+        }
+
+        /// <summary>
+        /// Forms a slice out of the given <see cref="ReadOnlyBuffer{T}"/>, beginning at <paramref name="offset"/>, and is at most <paramref name="length"/> bytes
+        /// </summary>
+        /// <param name="offset">The index at which to begin this slice.</param>
+        /// <param name="length">The length of the slice</param>
+        public ReadOnlyBuffer<T> Slice(long offset, long length)
+        {
+            var begin = Seek(BufferStart, BufferEnd, offset, false);
+            var end = Seek(begin, BufferEnd, length, false);
+            return SliceImpl(begin, end);
+        }
+
+        /// <summary>
+        /// Forms a slice out of the given <see cref="ReadOnlyBuffer{T}"/>, beginning at <paramref name="offset"/>, ending at <paramref name="end"/> (inclusive).
+        /// </summary>
+        /// <param name="offset">The index at which to begin this slice.</param>
+        /// <param name="end">The end (inclusive) of the slice</param>
+        public ReadOnlyBuffer<T> Slice(long offset, SequencePosition end)
+        {
+            BoundsCheck(BufferEnd, end);
+            var begin = Seek(BufferStart, end, offset);
+            return SliceImpl(begin, end);
+        }
+
+        /// <summary>
+        /// Forms a slice out of the given <see cref="ReadOnlyBuffer{T}"/>, beginning at <paramref name="start"/>, and is at most <paramref name="length"/> bytes
+        /// </summary>
+        /// <param name="start">The starting (inclusive) <see cref="SequencePosition"/> at which to begin this slice.</param>
+        /// <param name="length">The length of the slice</param>
+        public ReadOnlyBuffer<T> Slice(SequencePosition start, long length)
+        {
+            BoundsCheck(BufferEnd, start);
+
```

Comment:
nit: why newlines here but not in other places.

Reply:
Added newlines between BoundsCheck and rest of methods in other methods.